R/derive_param_confirmed_resp.R

```diff
@@ -294,13 +284,31 @@ derive_param_confirmed_resp <- function(dataset,
     assert_param_does_not_exist(dataset, quo_get_expr(set_values_to$PARAMCD))
   }
 
-  # Restrict input dataset
-  source_data <- dataset %>%
-    filter_pd(
-      filter = !!filter_source,
-      source_pd = source_pd,
-      source_datasets = source_datasets
-    )
+  #++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  # filter_pd and filter_source: Filter source dataset using filter_source----
+  # argument and also filter data after progressive disease with filter_pd
+  #++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+  if (!is.null(source_pd)) {
+    # Restrict input dataset
+    source_data <- dataset %>%
+      filter_pd(
+        filter = !!filter_source,
+        source_pd = source_pd,
+        source_datasets = source_datasets
```

Comment:
@sgorm123 also reminder to update for https://github.com/pharmaverse/admiralonco/issues/140 and you could add this issue to the Development section on RHS of this PR too if you do.

Reply:
ah yes, #140 is sorted, but somehow the fix is already on the development branch (I noticed this only today):

https://github.com/pharmaverse/admiralonco/blob/devel/R/derive_param_confirmed_resp.R#:~:text=subject_keys%20%3D%20subject_keys

Reply:
it doesn't seem to be fixed in the `filter_pd` call as per the issue detail - please double check